### PR TITLE
Added SecondaryStructure

### PR DESCRIPTION
**Pull request by anhi**

### CODE REVIEW

File: src/core/secondary_structure_type.jl [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using EnumX
```
Comment (tkemmer) [inline src/core/secondary_structure_type.jl:1]: Should be redundant with `using EnumX` being included in the main file already.

File: src/core/secondary_structure_type.jl [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+export SecondaryStructureElement, SecondaryStructureType
```
Comment (tkemmer) [inline src/core/secondary_structure_type.jl:3]:
```julia
export
    SecondaryStructureElement
    SecondaryStructureType
```

Comment (anhi) [reply src/core/secondary_structure_type.jl]: good point, i've changed that

File: src/core/secondary_structure.jl [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+ - `chain_idx::Int`
+
+# Constructors
+```julia
+SecondaryStructure(
+    chain::Chain{T};
+    number::Int,
+    type::SecondaryStructureType;
+    # keyword arguments
+    name::String,
```
Comment (tkemmer) [inline src/core/secondary_structure.jl:33]: `name::String = ""`

File: test/core/test_secondary_structure.jl [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+            1, 
+            SecondaryStructureElement.Helix; 
+            name = "H1",
+            properties = Properties(:first => 'a', :second => "b"),
+            flags = Flags([:third])
+        )
+        ss2 = SecondaryStructure(c, 2, SecondaryStructureElement.Coil; name="C1") 
+
+        st = secondary_structures(sys)
+
```
Comment (tkemmer) [inline test/core/test_secondary_structure.jl:21]: Additional tests introduced in d60011bf78df26d7a3d118b92582ab198e3ec6ff

```
        # AutoHashEquals and identity
        st2 = secondary_structures(sys)
        @test st == st2
        @test isequal(st, st2)
        @test hash(st) == hash(st2)
        @test st !== st2
```

**Pull request merged**